Framework/Core/include/Framework/DataAllocator.h

```diff
@@ -343,16 +358,17 @@ class DataAllocator
   RootObjectContext* mRootContext;
 
   std::string matchDataHeader(const Output &spec, size_t timeframeId);
-  FairMQMessagePtr headerMessageFromOutput(Output const &spec,
-                                           std::string const &channel,
-                                           o2::header::SerializationMethod serializationMethod);
+  FairMQMessagePtr headerMessageFromOutput(Output const& spec,                                  //
+                                           std::string const& channel,                          //
+                                           o2::header::SerializationMethod serializationMethod, //
+                                           size_t payloadSize);                                 //
 
-  Output getOutputByBind(OutputRef const& ref)
+  Output getOutputByBind(OutputRef&& ref)
   {
     for (size_t ri = 0, re = mAllowedOutputRoutes.size(); ri != re; ++ri) {
       if (mAllowedOutputRoutes[ri].matcher.binding.value == ref.label) {
         auto spec = mAllowedOutputRoutes[ri].matcher;
-        return Output{ spec.origin, spec.description, ref.subSpec, spec.lifetime };
+        return Output{ spec.origin, spec.description, ref.subSpec, spec.lifetime, std::move(ref.headerStack) };
```

Comment:
@ktf the move assignment in this line is the reason for the change to move semantics for the `OutputRef` arguments

Reply:
ok, but why does it need to be mutable?

Reply:
that said, doesn't this basically invalidate the OutputRef once it is moved?

Reply:
Stack does not support copy assignment, using move assignment leaves the original `OutputRef` rvalue with an empty Stack. But its a temporary object anyhow.

Reply:
the compiler does not take an lvalue at the function call, one has to explicitly apply `std::move` on the lvalue object. That should be enough to indicate that the object is not expected to be const at the function call. rarely to be noticed as we want to specify the `Outputref` in the call directly.

"Why not const", hmm, I would not expect it to be const if the Stack content is moved. Right now, the corresponding argument in `Output` is not const (for the same reason), so it does not compile with `OutputRef const&&` (discards qualifiers). I can play with the constness of the constructor argument, but I do not really expect it to work.

Reply:
I guess the bottom line is how this is going to be used. If people will create one `OutputRef` and then slightly modify it (e.g. to update the subspecification) or if the plan is to create a new `OutputRef` per invocation. The tradeoff depends on wether the creation of the Header stack is expensive or not I guess, although even in that case one could argue why do you want to optimise it then. Anyways, even if I am not sure this API is the correct one, we lack experience with it to really be able to tell if it's a good idea or not, so I agree to merge this and rediscuss later, if this brings things forward. We can always make it behave via copy later on.

Reply:
I agree, the API is probably not optimal and we do not have enough experience right now how it is going to be used. At the moment the question whether to copy or not is a design question if not academic, on the other hand we do not have a size limitation on the stack size, moving the stuff is the right thing to do in that case.

For the moment, the API is consistent and type safe. Furthermore, the case you mention can be implemented without lvalue object.